src/grammar.lalrpop

```diff
@@ -484,6 +482,15 @@ BinOp8: BinaryOp = {
 InfixExpr8: RichTerm = {
     InfixExpr7,
     LeftOp<BinOp8, InfixExpr8, InfixExpr7> => <>,
+    <l: @L> <t1: InfixExpr8> "!=" <t2: InfixExpr7> <r: @R> => {
```

Comment:
Any reason why the != is special-cased (compared to `=` for example)?

Reply:
`!=` isn't a primitive operator per se, but is parsed as `! (t1 == t2)`, which doesn't fit in the `LeftOp`/`InfixExpr` rules that rely on subrules like `BinOp8` to return a binary operator, and not an arbitrary expression.

The position (`@L` and `@R`) is needed if we want to pass it to the generated subexpression `t1 == t2`, which is set to have the same position as the original one (as well as the whole `! (t1 == t2)`). Usually, the more position information, the better.

However, thinking about this specific case a bit more, I don't think this position won't ever be useful: any typechecking error or evaluation error should be either located in a subterm `t1` or `t2`, or be about the whole expression `!(t1 == t2)`. Same for evaluation, if the whole expression is evaluated, `t1 == t2` will be evaluated away and the final position will be inherited from the enclosing `!`. 

The sub-expression `t1 == t2` is probably invisible for all intent of purpose, so maybe I should get rid of the `<l : @L>` and `<r: @R>`.